### PR TITLE
PP-10209 your PSP enable task list flag cypress test

### PR DESCRIPTION
**Pull request by AbdirizakIdris**
-Added cypress test for when ENABLE_STRIPE_ONBOARDING_TASK_LIST is set to true and also false:

### CODE REVIEW

File: test/cypress/integration/settings/switch-psp-onboarding-cy.test.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            { payment_provider: 'stripe', state: 'CREATED' }
+          ]),
+          stripeAccountSetupStubs.getGatewayAccountStripeSetupSuccess({
+            gatewayAccountId
+          })
+        ])
+      })
+      afterEach(()=>{process.env.ENABLE_STRIPE_ONBOARDING_TASK_LIST = false})
+      it('shows Stripe specific tasks', () => {
+        cy.visit(`/account/${gatewayAccountExternalId}/switch-psp`)
```
Comment (kbottla) [inline test/cypress/integration/settings/switch-psp-onboarding-cy.test.js:55]: This is still visiting `switch-psp` but changes should be tested for `your-psp` page

File: test/cypress/integration/settings/switch-psp-onboarding-cy.test.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      cy.get('.service-info--tag').should('not.contain', 'switching psp')
+      cy.get('#navigation-menu-switch-psp').should('have.length', 0)
+    })
+  })
+
+
+  describe('When stripe onboarding is switched to ', () => {
+    describe('Switching page', () => {
+      beforeEach(() => {
+        process.env.ENABLE_STRIPE_ONBOARDING_TASK_LIST = true
```
Comment (kbottla) [inline test/cypress/integration/settings/switch-psp-onboarding-cy.test.js:42]: For cypress tests, we can't test env variable within tests. Environment variable should be added in https://github.com/alphagov/pay-selfservice/blob/master/test/cypress/test.env

Also can't test for both false/true. So for cypress, we can enable the flag and check it.

File: test/cypress/integration/settings/switch-psp-onboarding-cy.test.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+
+function getUserAndAccountStubs (paymentProvider , providerSwitchEnabled, gatewayAccountCredentials, merchantDetails,) {
+  return [
+    userStubs.getUserSuccess({ gatewayAccountId, userExternalId, merchantDetails }),
+    gatewayAccountStubs.getGatewayAccountByExternalIdSuccess({ gatewayAccountId, gatewayAccountExternalId, providerSwitchEnabled, paymentProvider, ...gatewayAccountCredentials && { gatewayAccountCredentials } })
+  ]
+}
+
+describe('Switch PSP settings page', () => {
```
Comment (kbottla) [inline test/cypress/integration/settings/switch-psp-onboarding-cy.test.js:19]: This is not a switch PSP (we are only following the pattern). This should be named after the functionality being tested

**Review by kbottla — CHANGES_REQUESTED** [review]

File: test/cypress/integration/settings/your-psp-stripe-tasklist.cy.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  })
+
+  const stubs = [user, gatewayAccountByExternalId, stripeAccountSetup]
+
+  cy.task('setupStubs', stubs)
+}
+
+describe('Your PSP Stripe page', () => {
+    beforeEach(() => {  
+  Cypress.Cookies.preserveOnce('session', 'gateway_account')
```
Comment (kbottla) [inline test/cypress/integration/settings/your-psp-stripe-tasklist.cy.test.js:42]: Need a few more spaces 
```suggestion
    Cypress.Cookies.preserveOnce('session', 'gateway_account')
```

File: test/cypress/integration/settings/your-psp-stripe-tasklist.cy.test.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    type: 'live',
+    paymentProvider: 'stripe',
+    gatewayAccountCredentials: [{
+      payment_provider: 'stripe',
+      credentials: { stripe_account_id: stripeAccountId },
+      external_id: credentialExternalId
+    }]
+  })
+
+  const stripeAccountSetup = stripeAccountSetupStubs.getGatewayAccountStripeSetupSuccess({
```
Comment (kbottla) [inline test/cypress/integration/settings/your-psp-stripe-tasklist.cy.test.js:31]: Is this still required ?

Comment (AbdirizakIdris) [reply test/cypress/integration/settings/your-psp-stripe-tasklist.cy.test.js:31]: without tit he test fails  with a server error

File: test/cypress/integration/settings/your-psp-stripe-tasklist.cy.test.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    gatewayAccountId
+  })
+
+  const stubs = [user, gatewayAccountByExternalId, stripeAccountSetup]
+
+  cy.task('setupStubs', stubs)
+}
+
+describe('Your PSP Stripe page', () => {
+    beforeEach(() => {
```
Comment (kbottla) [inline test/cypress/integration/settings/your-psp-stripe-tasklist.cy.test.js:40]: There is still a bit of formatting that is off here. Fine to fix in the next PR.

**Review by kbottla — APPROVED** [review]

**Pull request merged**